hat/assets/js/apps/Iaso/domains/orgUnits/components/orgUnitMap/OrgUnitMap/OrgUnitMap.tsx

```diff
@@ -0,0 +1,568 @@
+            setStateField(keyName, {
+                ...state[keyName].value,
+                edit: !editEnabled,
+            });
+        },
+        [state, setStateField],
+    );
+
+    const toggleAddShape = useCallback(
+        keyName => {
+            const addEnabled = state[keyName].value.add;
+            if (addEnabled) {
+                const group =
+                    keyName === 'location'
+                        ? state.locationGroup.value
+                        : state.catchmentGroup.value;
+                group.shapeAdded.disable();
+            }
+            setStateField(keyName, {
+                ...state[keyName].value,
+                add: !addEnabled,
+            });
+        },
+        [state, setStateField],
+    );
+
+    const toggleDeleteShape = useCallback(
+        keyName => {
+            const deleteEnabled = state[keyName].value.delete;
+            const leafletMap = map.current.leafletElement;
+            const group =
+                keyName === 'location'
+                    ? state.locationGroup.value
+                    : state.catchmentGroup.value;
+            group.toggleDeleteShape(leafletMap, !deleteEnabled);
+            setStateField(keyName, {
+                ...state[keyName].value,
+                delete: !deleteEnabled,
+            });
+        },
+        [state, setStateField],
+    );
+
+    const addShape = useCallback(
+        keyName => {
+            const leafletMap = map.current.leafletElement;
+            if (keyName === 'location') {
+                state.locationGroup.value.addShape(leafletMap, 'primary');
+            }
+            if (keyName === 'catchment') {
+                state.catchmentGroup.value.addShape(leafletMap, 'secondary');
+            }
+            toggleAddShape(keyName);
+        },
+        [state.catchmentGroup.value, state.locationGroup.value, toggleAddShape],
+    );
+
+    const handleFormFilter = useCallback(
+        forms => {
+            setStateField('formsSelected', forms);
+        },
+        [setStateField],
+    );
+
+    const updateOrgUnitLocation = useCallback(
+        newOrgUnit => {
+            if (newOrgUnit.latitude && newOrgUnit.longitude) {
+                onChangeLocation({
+                    lat: newOrgUnit.latitude,
+                    lng: newOrgUnit.longitude,
+                    alt: newOrgUnit.altitude,
+                });
+            } else if (newOrgUnit.has_geo_json) {
+                setOrgUnitLocationModified(false); // What's the right value, the initial code passed nothing
+                onChangeShape('geo_json', newOrgUnit.geo_json);
+            }
+        },
+        [onChangeShape, onChangeLocation, setOrgUnitLocationModified],
+    );
+
+    const hasMarker =
+        Boolean(currentOrgUnit.latitude) && Boolean(currentOrgUnit.longitude);
+
+    if (map.current) {
+        map.current.leafletElement.options.maxZoom = currentTile.maxZoom;
+    }
```

Comment:
this is one is a bit weird... i assume it will disappear while upgrading leaflet

Reply:
That's what I assumed as well